sound/soc/sof/control.c

```diff
@@ -11,8 +11,41 @@
 /* Mixer Controls */
 
 #include <linux/pm_runtime.h>
+#include <linux/leds.h>
 #include "sof-priv.h"
 
+static void update_acpi_mute_led(struct snd_sof_control *scontrol,
+				 struct snd_kcontrol *kcontrol,
+				 struct snd_ctl_elem_value *ucontrol)
+{
+	struct snd_sof_dev *sdev = scontrol->sdev;
+	unsigned int mask;
+	unsigned int val;
+
+	mask = 1U << snd_ctl_get_ioffidx(kcontrol, &ucontrol->id);
+
+	if (ucontrol->value.integer.value[0] ||
+	    ucontrol->value.integer.value[1])
```

Comment:
@juimonen are you assuming stereo channels here? Should this be checked for scontrol->num_channels instead?

Reply:
@ranj063 see update, did you mean something like that?